content/300-guides/300-upgrade-guides/200-upgrading-versions/600-upgrading-to-prisma-5.mdx

```diff
@@ -0,0 +1,677 @@
+
+</Admonition>
+
+### TypeScript minimum version change
+
+From Prisma version 5.0.0, the minimum version of TypeScript supported is 4.7. If your project uses an earlier version of TypeScript, you will need to upgrade it.
+
+### PostgreSQL minimum version change
+
+From Prisma version 5.0.0, the minimum version of PostgreSQL supported is 9.6. If your project uses an earlier version of PostgreSQL, you will need to upgrade it.
+
+### Prisma Client embedded SQLite version updated
+
+With Prisma version 5.0.0, we have upgraded the embedded version of SQLite from `3.35.4` to `3.41.2`. We did not see any breaking changes and don't anticipate any changes needed in user projects, but if you are using SQLite, especially with raw queries that might go beyond Prisma's functionality, make sure to check [the SQLite changelog](https://www.sqlite.org/changes.html)
+
+## Primary changes
+
+This section gives an overview of the main breaking changes in Prisma 5.
+
+### Removal of <inlinecode>rejectOnNotFound</inlinecode> parameter
+
+From Prisma version 5.0.0, the deprecated parameter `rejectOnNotFound` has been removed. Depending on if your project used `rejectOnNotFound` per query or globally, there will be be different ways of updating your code.
+
+If you are using the `rejectOnNotFound` parameter on a per-query basis, then follow our steps for [updating your code at the query level](#replacing-rejectonnotfound-enabled-at-the-query-level)
+
+If instead you have set up the `rejectOnNotFound` parameter at the client level, you will need to follow [the steps for updating your code at the client level](#replacing-rejectonnotfound-enabled-at-the-client-level)
+
+#### Replacing <inlinecode>rejectOnNotFound</inlinecode> enabled at the query level.
+
+If you enable `rejectOnNotFound` on a per-query basis, you will need to replace your usage at the query level.
+
+The following example:
+
+```js
+prisma.user.findFirst({
+  where: { name: 'Alice' },
+  rejectOnNotFound: true,
+})
+```
+
+Is converted to:
+
+```js
+prisma.user.findFirstOrThrow({
+  where: { name: 'Alice' },
+})
+```
+
+If you use a custom error handler like the following:
+
+```js
+prisma.user.findFirst({
+  where: { name: 'Alice' },
+  rejectOnNotFound: () => new UserNotFoundError(),
+})
+```
+
+You will need to modify your code to handle the errors thrown by `...OrThrow` methods.
+
+```js
+try {
+  prisma.user.findFirstOrThrow({
+    where: { name: 'Alice' },
+  })
+} catch (err) {
+  if (err.code === 'P2025') {
+    throw new UserNotFoundError()
+  }
+  throw err
+}
+```
+
+You could also create a reusable error adapter which can be used in a `catch` in your code.
```

Comment:
What does this make sense? When they use the same custom error handler at many places in their code?